Android/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+cmake_minimum_required (VERSION 2.8.7)
+project(MCServer)
+include(ExternalProject)
+
+externalproject_add(lua_native
+	SOURCE_DIR ../lib/lua
+	CMAKE_GENERATOR "Visual Studio 14 2015"
+	CMAKE_ARGS -DANDROID_CROSSCOMPILE=TRUE
+	BUILD_COMMAND
+		${CMAKE_COMMAND} --build <BINARY_DIR> --config Release
+	INSTALL_COMMAND ""
+)
+externalproject_add(tolua_native
+	SOURCE_DIR ../lib/tolua++
+	CMAKE_GENERATOR "Visual Studio 14 2015"
+	BUILD_COMMAND
+		${CMAKE_COMMAND} --build <BINARY_DIR> --config Release
+	INSTALL_COMMAND ""
+)
+
+# Tell libevent that Android has pthread (not auto detected due to NDK implementation problems)
+# Tell Lua to not use unimplemented Android locale features
+add_definitions(-Dgetlocaledecpoint\(\)="'.'")
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14") # Add c++14 support here since we're not calling set_flags()
```

Comment:
Asking for c++14 can also result in broken code, if someone wants to copy android specific code to general codebase.